admin_guide/syncing_groups_with_ldap.adoc

```diff
@@ -0,0 +1,464 @@
+server as well as sync configuration options that are dependent on the schema used in your LDAP
+server.
+
+.LDAP Client Configuration
+====
+[source,yaml]
+----
+url: ldap://LDAP_SERVER_IP:389 <1>
+bindDN: cn=admin,dc=example,dc=com <2>
+bindPassword: admin <3>
+insecure: true <4>
+ca: my-ldap-ca-bundle.crt <5>
+----
+<1> The connection protocol, IP address of the LDAP server hosting your database, and the port to
+connect to, formatted as `scheme://host:port`.
+<2> The optional DN to use to bind if elevated privilege is necessary to retrieve entries for the
+sync job.
+<3> The optional password to use to bind if elevated privilege is necessary to retrieve entries for
+the sync job.
+<4> When *true*, no TLS connection is made to the server. When *false*, `ldaps://` URLs connect
+using TLS, and `ldap://` URLs are upgraded to TLS.
+<5> Certificate bundle to use to validate server certificates for the configured URL. If empty,
+system trusted roots are used. Only applies if *insecure: false*.
+====
+
+Sync configurations consist of LDAP query definitions for all of the entries that are
+required for the sync job. The specific definition of an LDAP query will depend on the schema used
+to store membership information in the LDAP server.
+
+.LDAP Query Definition
+====
+[source,yaml]
+----
+baseDN: ou=users,dc=example,dc=com <1>
+scope: sub <2>
+derefAliases: never <3>
+timeout: 0 <4>
+filter: (objectClass=intOrgPerson) <5>
+----
+<1> The DN of the branch of the directory where all searches should start from. At the very least,
+this must be the top of your directory tree, but it could also specify a subtree in the directory.
+<2> The scope of the search. Can be either either `base`, `one` or `sub`. If the scope is not
+provided, the default is to use a scope of `sub`. Descriptions of the scope options are given in the
+link:#ldap-search[appendix.]
+<3> The behavior of the search with respect to aliases in the LDAP tree. Can be either `never`,
+`search`, `base` or `always`. If the behavior is not given, the default is to `always` dereference
+aliases. Descriptions of the dereferencing behaviors are given in the
+link:#deref-aliases[appendix].
+<4> The time limit allowed for the search by the client, in seconds. A value of 0 imposes no client-
+side limit.
+<5> A valid LDAP search filter. If not provided, defaults to `(objectClass=*)`.
+====
+
+== Running an LDAP Sync Job
+Once a configuration file is created for the sync job, the sync job can be started. OpenShift allows
+an administrator to perform a number of different sync types with the same server:
+
+An administrator can sync all groups from the LDAP server into OpenShift:
+----
+$ openshift ex sync-groups --sync-config=config.yaml
+----
+An administrator can sync all groups already in OpenShift that correspond to the server specified
+in the configuration file:
+----
+$ openshift ex sync-groups --type=openshift --sync-config=config.yaml
```

Comment:
I've re-written that section, hopefully for the better. I'm thinking about making a table or something, it's not the clearest command.